### PR TITLE
Fix `addSpnToEnvironment` casing bug due to difference in casing

### LINKED ISSUE #12933: Azure `addSpnToEnvironment` has a casing bug which causes environment variables to not be published
## Required Information

Entering this information will route you directly to the right team and expedite traction.

**Question, Bug, or Feature?**  
*Type*: BUG

**Enter Task Name**: AzureCLI Task  

list here:  
https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/AzureCLIV2/azureclitask.ts#L112

## Environment
- Server - Azure Pipelines
- Agent - Hosted or Private: 
    
```
2020-05-19T07:54:03.8267565Z Agent name: 'Azure Pipelines 2'
2020-05-19T07:54:03.8267929Z Agent machine name: 'fv-az100'
2020-05-19T07:54:03.8268161Z Current agent version: '2.168.2'
2020-05-19T07:54:03.8319171Z ##[group]Operating System
2020-05-19T07:54:03.8319460Z Ubuntu
2020-05-19T07:54:03.8319613Z 18.04.4
2020-05-19T07:54:03.8319758Z LTS
2020-05-19T07:54:03.8320108Z ##[endgroup]
2020-05-19T07:54:03.8320314Z ##[group]Virtual Environment
2020-05-19T07:54:03.8320567Z Environment: ubuntu-18.04
2020-05-19T07:54:03.8320788Z Version: 20200512.2
2020-05-19T07:54:03.8321164Z Included Software: https://github.com/actions/virtual-environments/blob/ubuntu18/20200512.2/images/linux/Ubuntu1804-README.md
2020-05-19T07:54:03.8321506Z ##[endgroup]
2020-05-19T07:54:03.8322588Z Current image version: '20200512.2'
2020-05-19T07:54:03.8326992Z Agent running as: 'vsts'
2020-05-19T07:54:03.8336693Z Prepare release directory.
```

## Issue Description

The `addSpnToEnvironment` check is missing a `toLower`. We observe that service connection create manually have a `authentication.Scheme = serviceprincipal` whereas those created using different approach have  `authentication.Scheme = ServicePrincipal`

Here are screen grabs showing the different casing from two service connections we have in our setup.

image

image

Check which asserts casing is correct during `addSpnToEnvironment`:

https://github.com/microsoft/azure-pipelines-tasks/blob/ea135f8b9f16fe26f8e232abe1efa9039910384d/Tasks/AzureCLIV2/azureclitask.ts#L43-L49

This is contrasted with the code in the login section which **does** do the `tolower` check:

https://github.com/microsoft/azure-pipelines-tasks/blob/ea135f8b9f16fe26f8e232abe1efa9039910384d/Tasks/AzureCLIV2/azureclitask.ts#L111-L114

### PR DESCRIPTION
**Pull request by lawrencegripper**
Fixes #12933

### CODE REVIEW

**Comment by lawrencegripper** [comment]
I've checked this out and build it. Looks like there isn't a test suite with real tests for the task.

**Comment by bishal-pdMSFT** [comment]
Thanks for the PR @lawrencegripper 
For building and running tests, please see the [contribution guide](https://github.com/microsoft/azure-pipelines-tasks/blob/master/docs/contribute.md)
For uploading task to your ADO organization and test its functionality, please use [tfx cli](https://github.com/microsoft/tfs-cli/blob/master/docs/buildtasks.md)

**Comment by lawrencegripper** [comment]
Thanks @bishal-pdMSFT - I've run the build and tests. Just wanted to call out that the tests for this task are "hello world".

image

image

File: Tasks/AzureCLIV2/azureclitask.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export class azureclitask {
             });
 
             var addSpnToEnvironment: boolean = tl.getBoolInput("addSpnToEnvironment", false);
-            if (!!addSpnToEnvironment && tl.getEndpointAuthorizationScheme(connectedService, true) == "ServicePrincipal") {
+            if (!!addSpnToEnvironment && tl.getEndpointAuthorizationScheme(connectedService, true).toLowerCase() == "serviceprincipal") {
```
Comment (bishal-pdMSFT) [inline Tasks/AzureCLIV2/azureclitask.ts:44]: @lawrencegripper please bump up the task patch version

Comment (lawrencegripper) [reply Tasks/AzureCLIV2/azureclitask.ts:44]: Sure, bumped to 2.0.13

**Pull request merged**